common/src/main/java/com/gentics/mesh/cache/impl/EventAwareCacheImpl.java

```diff
@@ -35,7 +38,7 @@ public class EventAwareCacheImpl<K, V> implements EventAwareCache<K, V> {
 
 	private static final Logger log = LoggerFactory.getLogger(EventAwareCacheImpl.class);
 
-	private final Cache<K, V> cache;
+	private final Cache<K, Optional<V>> cache;
```

Comment:
Most implementation of AbstractMeshCache (which internally uses EventAwareCache) will not clean the cache upon creation of elements. This means that cached empty Optionals will not be invalidated properly.
We definitely need test cases for ALL implementations for (example made for project name):
1. Load (inexistent) project by name (will now put an empty Optional into the cache)
2. Create project by that name
3. Try to load the project with the name again (should find the project now)

Reply:
Added the reaction over `*CREATED` events. The unit test set is TBD.

Reply:
UTs added.